models/issue_list.go

```diff
@@ -0,0 +1,366 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import "fmt"
+
+// IssueList defines a list of issues
+type IssueList []*Issue
+
+func (issues IssueList) getRepoIDs() []int64 {
+	repoIDs := make([]int64, 0, len(issues))
```

Comment:
I would suggest doing something like

```
repoIDs := make(map[int64]bool, len(issues))
for _, issue := range issues {
  repoIDs[issue.RepoID] = true
}
return keys(repoIDs) // keys is a helper function (map[int64]bool) -> []int64
```

This is asymptotically more efficient, and more concise (since the `keys` helper function can be shared across the `getSomethingIDs` functions)